### PR TITLE
Replacing akka with pekko.

### LINKED ISSUE #641: Replace akka with pekko
According to [this guide](https://pekko.apache.org/docs/pekko/current/project/migration-guides.html), it seems straight forward

### PR DESCRIPTION
**Pull request by staudtMarius**
Resolves #641

### CODE REVIEW

**Comment by sebastian-peter** [comment]
@staudtMarius With the release of simonaAPI 0.4.0, your PR is included, so it should fix this PR once you update the dependency

File: src/test/scala/edu/ie3/simona/agent/grid/DBFSAlgorithmCenGridSpec.scala [hunk 1]
```diff
@@ -53,8 +53,8 @@ class DBFSAlgorithmCenGridSpec
         "DBFSAlgorithmCenGridSpec",
         ConfigFactory
           .parseString("""
-            |akka.loggers =["akka.event.slf4j.Slf4jLogger"]
-            |akka.loglevel="OFF"
+            |org.apache.pekko.loggers =["org.apache.pekko.event.slf4j.Slf4jLogger"]
+            |org.apache.pekko.loglevel="OFF"
```
Comment (sebastian-peter) [inline src/test/scala/edu/ie3/simona/agent/grid/DBFSAlgorithmCenGridSpec.scala:57]: We found out this doesn't work, right? Please check for all tests

File: src/test/scala/edu/ie3/simona/config/ConfigFailFastSpec.scala [hunk 2]
```diff
@@ -948,55 +948,55 @@ class ConfigFailFastSpec extends UnitSpec with ConfigTestData {
...
           /* Testing the log message cannot be tested, as with LazyLogging, the logger of the class cannot be spied. */
         }
 
         "pass on proper config" in {
           val properConfig = ConfigFactory.parseString(
             """
-              |akka {
-              |  loggers = ["akka.event.slf4j.Slf4jLogger"]
+              |pekko {
+              |  loggers = ["pekko.event.slf4j.Slf4jLogger"]
```
Comment (sebastian-peter) [inline src/test/scala/edu/ie3/simona/config/ConfigFailFastSpec.scala:967]: ... like this!

File: build.gradle [hunk 3]
```diff
@@ -123,14 +123,15 @@ dependencies {
...
-  implementation "com.lightbend.akka:akka-stream-alpakka-csv_${scalaVersion}:4.0.0"
-  implementation "com.typesafe.akka:akka-cluster-sharding_${scalaVersion}:${akkaVersion}"
-  implementation "com.typesafe.akka:akka-cluster-tools_${scalaVersion}:${akkaVersion}"
+  /* CORE Pekko */
+  implementation "org.apache.pekko:pekko-actor_${scalaVersion}:${pekkoVersion}"
+  implementation "org.apache.pekko:pekko-actor-typed_${scalaVersion}:${pekkoVersion}"
+  implementation "org.apache.pekko:pekko-slf4j_${scalaVersion}:${pekkoVersion}"
+  implementation "org.apache.pekko:pekko-cluster_${scalaVersion}:${pekkoVersion}"
+  implementation "org.apache.pekko:pekko-connectors-csv_${scalaVersion}:${pekkoVersion}"
+  implementation "org.apache.pekko:pekko-connectors-kafka_${scalaVersion}:1.0.0"
```
[the remark this answers is not in the document]
Comment (staudtMarius) [reply build.gradle:132]: Before we used `com.lightbend.akka:akka-stream-alpakka-csv`, I replaced it with the `pekko` components.
If the kafka connector is irelevant I could remove it.

[the remark this answers is not in the document]
Comment (sebastian-peter) [reply build.gradle]: There's still one `cluster` dependency

[the remark this answers is not in the document]
Comment (sebastian-peter) [reply build.gradle]: Generally, remove everything if SIMONA still compiles (and all tests succeed)

[the remark this answers is not in the document]
Comment (staudtMarius) [reply build.gradle]: I removed the clusters completly.

**Comment by staudtMarius** [comment]
!test

File: input/samples/common/pekko.conf [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  loggers = ["org.apache.pekko.event.slf4j.Slf4jLogger"]
+  # Options: OFF, ERROR, WARNING, INFO, DEBUG
+  loglevel = "debug"
+  logging-filter = "org.apache.pekko.event.slf4j.Slf4jLoggingFilter"
+
+  actor {
+
+    // pekko supervision strategy for failure handling
+    // https://pekko.apache.org/docs/pekko/current/typed/fault-tolerance.html
+    // and https://pekko.apache.org/docs/pekko/current/general/supervision.html#user-guardian
```
Comment (sebastian-peter) [inline input/samples/common/pekko.conf:12]: Thanks for looking up the equivalent doc links!

**Review by sebastian-peter — APPROVED** [review]

**Pull request merged**